### PR TITLE
[grafana] Add targetLabels to ServiceMonitors

### PR DESCRIPTION
**Pull request by jkroepke**

### CODE REVIEW

File: charts/grafana/values.yaml [hunk 1]
```diff
@@ -194,6 +194,7 @@ serviceMonitor:
   tlsConfig: {}
   scrapeTimeout: 30s
   relabelings: []
+  targetLabels: []
```
Comment (zanhsieh) [inline charts/grafana/values.yaml:197]: This should be empty map `{}`:
```suggestion
  targetLabels: {}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/grafana/values.yaml (hunk 1 above)
Comment (jkroepke) [reply charts/grafana/values.yaml:197]: Why? `targetLabels` is a list.

Comment (zanhsieh) [reply charts/grafana/values.yaml:197]: Then, did you provide an example in values.yaml? How does an engineer / developer know?

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/grafana/values.yaml (hunk 1 above)
Comment (jkroepke) [reply charts/grafana/values.yaml:197]: In the same way, how he knows set `relabelings`. I can provide list of string as example. Does it help?

```
% k explain servicemonitor.spec.targetLabels
KIND:     ServiceMonitor
VERSION:  monitoring.coreos.com/v1

FIELD:    targetLabels <[]string>

DESCRIPTION:
     TargetLabels transfers labels from the Kubernetes `Service` onto the
     created metrics.
```

Comment (zanhsieh) [reply charts/grafana/values.yaml:197]: Yes.

File: charts/grafana/values.yaml [hunk 2]
```diff
@@ -1068,6 +1069,7 @@ imageRenderer:
     tlsConfig: {}
     scrapeTimeout: 30s
     relabelings: []
+    targetLabels: []
```
Comment (zanhsieh) [inline charts/grafana/values.yaml:1072]:
```suggestion
    # See: https://doc.crds.dev/github.com/prometheus-operator/kube-prometheus/monitoring.coreos.com/ServiceMonitor/v1@v0.11.0#spec-targetLabels
    targetLabels: []
      # - targetLabel1
      # - targetLabel2
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

File: charts/grafana/templates/image-renderer-servicemonitor.yaml [hunk 3]
```diff
@@ -41,4 +41,8 @@ spec:
   namespaceSelector:
     matchNames:
       - {{ include "grafana.namespace" . }}
+  {{- with .Values.serviceMonitor.targetLabels }}
```
Comment (peikk0) [inline charts/grafana/templates/image-renderer-servicemonitor.yaml:44]: Should have been:
```
{{- with .Values.imageRenderer.serviceMonitor.targetLabels }}
```

Comment (peikk0) [reply charts/grafana/templates/image-renderer-servicemonitor.yaml:44]: Fix in https://github.com/grafana/helm-charts/pull/2137

**Pull request merged**